File: JPASquadGoals/src/main/java/com/skilldistillery/squadgoals/entities/Badge.java
package com.skilldistillery.squadgoals.entities;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
public class Badge {
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private int id;
	private String name;
	private String description;
	@ManyToMany(mappedBy="badges")
	@JsonIgnoreProperties({"badges"})
	private List<User> users;
	

	public Badge() {
		
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public List<User> getUsers() {
		return users;
	}

	public void setUsers(List<User> users) {
		this.users = users;
	}

	public void addUser(User user) {
		if (users == null) {
			users = new ArrayList<>();
		}
		if (!users.contains(user)) {
			users.add(user);
			user.addBadge(this);
		}
	}

	public void removeUser(User user) {
		if (users != null && users.contains(user)) {
			users.remove(user);
			user.removeBadge(this);
		}
	}
	
}

File: ngSquadGoals/src/app/models/badge.ts
export class Badge {
  id: number = 0;
  name: string = '';
  description: string = '';
  users: any[] = [];
  squads: any[] = [];
  requirements: any[] = [];

  constructor(
    id: number = 0,
  name: string = '',
  description: string = '',
  users: any[] = [],
  squads: any[] = [],
  requirements: any[] = []
  ) {
  this.id = id;
  this.name = name;
  this.description = description;
  this.users = users;
  this.squads = squads;
  this.requirements = requirements;
  }
}
